packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts

```diff
@@ -231,7 +235,7 @@ export function loadApiWithPrivacyParams(context: $TSContext, name: string, reso
 }
 
 function updateExistingApiCfn(context: $TSContext, api: $TSObject): void {
-  const { resourceName } = api.params;
+  const resourceName = api.resourceName || api.params.resourceName;
```

Comment:
should we have any defensive logic here to early return if resourceName is undefined?